runtime/src/test/java/com/tietoevry/quarkus/resteasy/problem/HttpProblemTest.java

```diff
@@ -0,0 +1,56 @@
+package com.tietoevry.quarkus.resteasy.problem;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.net.URI;
+import org.junit.jupiter.api.Test;
+import org.zalando.problem.Status;
+
+class HttpProblemTest {
+
+    @Test
+    void emptyProblemShouldHaveTypeAboutBlank() {
+        HttpProblem problem = HttpProblem.builder().build();
+
+        assertThat(problem.getType().toString()).isEqualTo("about:blank");
+    }
+
+    @Test
+    void builderShouldPassAllFields() {
+        HttpProblem problem = sampleHttpProblem();
+
+        assertThat(problem.getType())
+                .hasHost("tietoevry.com")
+                .hasPath("/problem");
+
+        assertThat(problem.getStatus().getStatusCode()).isEqualTo(400);
+        assertThat(problem.getDetail()).isEqualTo("This is detail");
+
+        assertThat(problem.getHeaders())
+                .containsEntry("X-Numeric-Header", 123)
+                .containsEntry("X-String-Header", "ABC");
+    }
+
+    @Test
+    void copyBuilderShouldCopyAllFields() {
+        HttpProblem original = sampleHttpProblem();
+
+        HttpProblem copy = HttpProblem.builder(original).build();
+
+        assertThat(copy)
+                .isEqualToComparingFieldByField(original);
```

Comment:
`.isEqualToComparingFieldByField` is deprecated?

Reply:
Nope, it's not, at least not in assertj version we're using.